### PR TITLE
CFE-3456/master: Refactored motd example

### PR DESCRIPTION
**Pull request by nickanderson**

### CODE REVIEW

File: examples/mustache_template_motd.cf [hunk 1]
```diff
@@ -1,71 +1,120 @@
...
-
-    "admin_contact"
-      slist => { "root@localhost" },
-      if => strcmp( $(environment), "unknown" );
-
-  ## We define a class to conditionally render the role status in the final
-  ## motd.
-  classes:
-    "env_$(environment)" expression => "any";
+      "motd_path" string => "/tmp/motd";
```
Comment (vpodzime) [inline examples/mustache_template_motd.cf:68]: Why not `/etc/motd`?

File: examples/mustache_template_motd.cf [hunk 2]
```diff
@@ -1,71 +1,120 @@
...
+
+      "template_data"
+        data => mergedata('{ "fqhost": "$(sys.fqhost)",
+                   "primary_ip": "$(sys.ipv4)",
+                   "cf_version": "$(sys.cf_version)",
+                   "issue_msg": "$(env_classification.env_issue_msg)",
+                   "cf_role":  "$(env_classification.cf_role)",
+                   "count_updates": "$(env_info.count_updates)",
+                   "contacts": env_info.admin_contact,
+                   "classes": _state[classes]
```
Comment (vpodzime) [inline examples/mustache_template_motd.cf:88]: Should use `c`, right? Or it should not be defined above.

File: examples/mustache_template_motd.cf [hunk 3]
```diff
@@ -1,71 +1,120 @@
...
 
-  ## In order to get package inventory, we need to have a legacy package
-  ## promise.
-  packages:
-    "vim" -> { "https://dev.cfengine.com/issues/7864" }
-      package_policy => "add";
+      # First we extract currently defined classes from datastate(), then we
+      # construct the template data.
+
+      "_state" data => datastate(), if => not( isvariable ( $(this.promiser) ) );
```
Comment (vpodzime) [inline examples/mustache_template_motd.cf:77]: What's the purpose of the `if => not(isvariable($(this.promiser)))`? I think it needs an explanation.

**Review by vpodzime — COMMENTED** [review]
Looks good to me otherwise. Thanks for working on this!

File: examples/mustache_template_motd.cf (hunk 3 above)
Comment (nickanderson) [reply examples/mustache_template_motd.cf:77]: ack, added explination.

**Comment by nickanderson** [comment]
Please re-review. if OK i will squash.

File: examples/mustache_template_motd.cf [hunk 4]
```diff
@@ -74,7 +74,12 @@ bundle agent motd {
       # First we extract currently defined classes from datastate(), then we
       # construct the template data.
 
-      "_state" data => datastate(), if => not( isvariable ( $(this.promiser) ) );
+      "_state" data => datastate(),
+        if => not( isvariable ( $(this.promiser) ) ); # Limit recursive growth
+                                                      # and multiple calls to
+                                                      # datastate() over
+                                                      # multiple passes.
```
Comment (olehermanse) [inline examples/mustache_template_motd.cf:81]: Nice!

**Comment by nickanderson** [comment]
Squashed.

**Review by craigcomstock — APPROVED** [review]
cool. looks like a nice refactor.

**Review by vpodzime — APPROVED** [review]
Looks good to me now, thanks!

**Pull request merged**